src/audio/dai.c

```diff
@@ -66,6 +93,55 @@ struct dai_data {
 	uint64_t wallclock;	/* wall clock at stream start */
 };
 
+static void dai_atomic_trigger(void *arg, enum notify_id type, void *data);
+
+static int dai_add_group(struct comp_dev *dev, uint8_t group_id)
+{
+	struct list_item *dai_groups = &groups[cpu_get_id()].list;
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct list_item *group_item;
+	struct dai_group *group;
+
+	if (dd->group) {
+		if (dd->group->group_id != group_id) {
+			comp_err(dev, "dai_add_group(), DAI already in group");
+			return -EINVAL;
+		}
+
+		/* No need to re-assign to the same group, do nothing */
+		return 0;
+	}
+
+	list_for_item(group_item, dai_groups) {
+		group = container_of(group_item, struct dai_group, list);
+
+		if (group->group_id == group_id)
+			break;
+
+		group = NULL;
+	}
+
+	if (!group) {
+		group = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM,
+				sizeof(*group));
+
+		list_item_prepend(&group->list, dai_groups);
+		group->group_id = group_id;
+		group->trigger_cmd = -1; /* no trigger command */
+	}
+
+	group->num_dais++;
+	dd->group = group;
+
+	comp_dbg(dev, "dai_add_group(), group %d num %d",
+		 group_id, group->num_dais);
+
+	notifier_register(dev, group, NOTIFIER_ID_DAI_TRIGGER,
```

Comment:
an inline comment stating what this notifier is used for

Reply:
Used to actually run the command on all components in the group when the last request came through.